### ISSUE TITLE
[REVIEW]: pyemgpipeline: A Python package for electromyography processing

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4156]
**Submitting author:** @tlwu (<a href="http://orcid.org/0000-0002-1823-6818">Tsung-Lin Wu</a>)
**Repository:** https://github.com/aalhossary/pyemgpipeline
**Branch with paper.md** (empty if default branch): 
**Version:** version 1.0.0_RC
**Editor:** @osorensen
**Reviewers:** @tuliofalmeida, @osorensen
**Archive:** 10.5281/zenodo.6422544

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8/status.svg)](https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8"><img src="https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8/status.svg)](https://joss.theoj.org/papers/c95cc7e641fc919b3ba3a040189f90d8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/aalhossary/pyemgpipeline" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@tuliofalmeida, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @osorensen know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @tuliofalmeida
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/aalhossary/pyemgpipeline">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/aalhossary/pyemgpipeline">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tlwu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Wordcount for `paper.md` is 700

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.jelekin.2016.10.001 is OK
- 10.1016/j.jelekin.2005.07.003 is OK
- 10.1016/S1050-6411(18)30348-1 is OK
- 10.21105/joss.01870 is OK

MISSING DOIs

- 10.1201/9781420036985.ax2 may be a valid DOI for title: Standards for reporting EMG data

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.22 s (380.1 files/s, 133364.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript                      13           2405           2470           9219
HTML                            13            435             39           5304
SVG                              1              0              0           2671
Python                          36            550            962           1586
CSS                              4            191             35            758
Jupyter Notebook                 3              0           1783            170
reStructuredText                 8             81            122            100
Markdown                         2             34              0             92
TeX                              1              5              0             54
DOS Batch                        1              8              1             26
make                             1              4              7              9
-------------------------------------------------------------------------------
SUM:                            83           3713           5419          19989
-------------------------------------------------------------------------------


Statistical information for the repository '94a06286cd553fda6967ed41' was
gathered on 2022/02/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Amr ALHOSSARY                    1           182              0            0.24
tlwu                            31         46050          28946           99.76

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Amr ALHOSSARY                 4            2.2          0.0                0.00
tlwu                      17282           37.5          1.1               15.72
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04156/joss.04156/10.21105.joss.04156.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04156/joss.04156/10.21105.joss.04156.pdf) :page_facing_up: :point_left:

**Comment by osorensen** [comment]
@tlwu can you please update us on how the work with the revision is going?

**Comment by tlwu** [comment]
@osorensen sorry that I am busy with other projects recently. I will complete the revision in two weeks. Thanks!

**Comment by osorensen** [comment]
That's fine @tlwu. Thanks for the notice:-)

**Comment by tuliofalmeida** [comment]
@osorensen The authors made the suggested changes. The software is working well and the examples are meaningful, easy to understand and interactive. I have completed my review.

**Comment by osorensen** [comment]
Thanks @tlwu. Your review is much appreciated.

**Comment by osorensen** [comment]
@editorialbot generate pdf

**Comment by osorensen** [comment]
@tlwu, the reviewer has now completed the checklist. While I go through the paper, could you please do the following?
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by osorensen** [comment]
@editorialbot check references

**Comment by osorensen** [comment]
@tlwu, I have opened Issues in the source repository with some small editorial comments. Please also check if the DOI suggest by the editorialbot in the post above should be added.

**Comment by tlwu** [comment]
> MISSING DOIs
> 
> - 10.1201/9781420036985.ax2 may be a valid DOI for title: Standards for reporting EMG data

This reference can be found by googling this DOI, but it cannot be found by searching the DOI at https://www.doi.org/. We've added this DOI in the paper.

**Comment by tlwu** [comment]
> @tlwu, I have opened Issues in the source repository with some small editorial comments. Please also check if the DOI suggest by the editorialbot in the post above should be added.

@osorensen, thank you for the editorial comments. The paper has been revised based on the editorial and DOI suggestions.

**Comment by tlwu** [comment]
>     * [ ]  Make a tagged release of your software, and list the version tag of the archived version here.

The software is released. The version tag is v1.0.0_RC.

>     * [ ]  Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
>     * [ ]  Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
>     * [ ]  Please list the DOI of the archived version here.

The software is archived in Zenodo, the metadata is checked, and the DOI of the archived version is 10.5281/zenodo.6422544.

**Comment by osorensen** [comment]
@editorialbot generate pdf

**Comment by osorensen** [comment]
@editorialbot set version v1.0.0_RC

**Comment by osorensen** [comment]
@editorialbot set version 1.0.0_RC as version

**Comment by osorensen** [comment]
@editorialbot set 10.5281/zenodo.6422544 as archive

**Comment by osorensen** [comment]
@editorialbot check references

**Comment by osorensen** [comment]
@editorialbot recommend-acceptance

**Comment by osorensen** [comment]
@editorialbot recommend-accept

**Comment by osorensen** [comment]
@tlwu, my apologies, but I've been a bit too quick here. Our policy is to have at least two reviewers, so since I was only able to find a single reviewer, I'll also do a review myself before we can proceed. I'll start working on the review tomorrow, and will let you know if I have any suggested changes.

**Comment by osorensen** [comment]
@editorialbot add @osorensen as reviewer

**Comment by osorensen** [comment]
@editorialbot generate my checklist

**Comment by osorensen** [comment]
@openjournals/jose-eics, I have now completed my review. The submitted software looks very complete, with good documentation and examples.

**Comment by osorensen** [comment]
@editorialbot generate pdf

**Comment by tlwu** [comment]
> @tlwu, my apologies, but I've been a bit too quick here. Our policy is to have at least two reviewers, so since I was only able to find a single reviewer, I'll also do a review myself before we can proceed. I'll start working on the review tomorrow, and will let you know if I have any suggested changes.

@osorensen thank you!

**Comment by osorensen** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@tuliofalmeida, @osorensen – many thanks for your reviews here and to @osorensen for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@tlwu – your paper is now accepted and published in JOSS :zap::rocket::boom:
